src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,132 @@
 package core.basesyntax;
 
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
+            throws DataGettingException {
+
+        if (names == null || data == null || dateFrom == null
+                || dateTo == null) {
+            throw new DataGettingException("one or more input arguments are null");
+        }
+
+        int dayFrom = 0;
+        int monthFrom = 0;
+        int yearFrom = 0;
+
+        int dayTo = 0;
+        int monthTo = 0;
+        int yearTo = 0;
+
+        try {
+            dayFrom = getNumberOfDay(dateFrom, "day");
+            monthFrom = getNumberOfDay(dateFrom, "month");
+            yearFrom = getNumberOfDay(dateFrom, "year");
+
+            dayTo = getNumberOfDay(dateTo, "day");
+            monthTo = getNumberOfDay(dateTo, "month");
+            yearTo = getNumberOfDay(dateTo, "year");
+        } catch (WrongWordException e) {
+            System.out.println("Input wrong word");
+        }
+
+        if (yearFrom > yearTo || (yearFrom == yearTo && monthFrom > monthTo)
+                || (yearFrom == yearTo && monthFrom == monthTo && dayFrom > dayTo)) {
```

Comment:
The logic for checking if the start date is later than the end date can be simplified using `LocalDate` comparisons.